collections/_posts/2023-07-06-Mean_teachers_are_better_role_models.md

```diff
@@ -0,0 +1,210 @@
+- Both predictions can be used but the **teacher prediction is more likely to be correct**
+- No classification cost is applied when it is a training step with unlabeled example. 
+
+### Consistency cost 
+
+With $$J$$ the consistency cost, as the expected distance between the prediction of the student model (with weights $$θ$$ and noise $$η$$) and the prediction of the teacher model (with weights $$θ'$$ and noise $$η'$$) :
+
+$$ J(θ) = \underset{x,η',η}{E} [ || f (x, θ', η') - f (x, θ, η) ||^2 ] $$
+
+Mean squared error is applied to the distance to get the consistency cost. The weight of this consistency loss changes during training: it ramps up from 0 to its final value during the first 80 epochs.
+ &rarr; Initially ignoring it and then gradually introducing more and more consistency. 
+
+### Difference between the $$\Pi$$ model, Temporal Ensembling, and Mean teacher 
+
+- How the teacher predictions are generated :
+    - $$\Pi$$ model : $$θ' = θ$$  
+    - TE : approximates $$f (x, θ', η')$$ with a **weighted average of successive predictions**
+    - Mean Teacher defines $$\underset{t}{θ'}$$ at training step as the **EMA of successive $$\theta$$ weights**, where $$\alpha$$ is a smoothing coefficient hyperparameter : 
+
+$$ \underset{t}{θ'} = α \underset{t-1}{θ'} + (1 - α)\underset{t}{θ} $$
+
+- Weights $$\theta'$$ :
+    - $$\Pi$$ model : applies training to it
+    - TE and Mean teacher : treat it as a constant with regards to **optimization**
+
+### Type of noise 
+- Random translations and horizontal flips of the input images
+- Gaussian noise on the input layer
+- Dropout applied within the network
+
+### Loss
+- Student : cross-entropy loss 
+- Teacher : consistency loss multiplied by an importance weight 
+
+## Data
+**Street View House Numbers (SVHN)** : 
+- 32x32 pixel RGB images belonging to ten different classes
+- close-up of a house number, and the class represents the identity of the digit at the center. 
+- 73,257 training samples and 26,032 test samples.
+
+![](/collections/images/mean_teachers/SVHN.jpg)
+
+**CIFAR-10** : 
+- 32x32 pixel RGB images belonging to ten different classes
+- natural image belonging to a class such as horses, cats, cars and airplanes etc
+- 50,000 training samples and 10,000 test samples.
+
+![](/collections/images/mean_teachers/CIFAR-10.jpg)
+
+# Results
+
+## Comparisons with the state-of-the-art methods:
+
+![](/collections/images/mean_teachers/table_1.jpg)
+
+**Mean Teacher improves test accuracy** over the $$\Pi$$ model and Temporal Ensembling on semi-supervised SVHN tasks. 
+
+![](/collections/images/mean_teachers/table_2.jpg)
+
+Mean Teacher also improves results on CIFAR-10 over our baseline $$\Pi$$ model.
+
+> **NOTE :** The Virtual Adversarial Training (VAT) by [Miyato et al. 2017](https://doi.org/10.48550/arXiv.1704.03976)[2] performs even better than Mean Teacher on the **1000-label SVHN** and the **4000-label CIFAR-10**. VAT and Mean Teacher are complimentary approaches.  
```

Comment:
Fix the already existing footnote rather than inline the reference:
```suggestion
> **NOTE :** The Virtual Adversarial Training (VAT) by Miyato et al. 2017[^2] performs even better than Mean Teacher on the **1000-label SVHN** and the **4000-label CIFAR-10**. VAT and Mean Teacher are complimentary approaches.  
```